backend/metamodel/syson-sysml-metamodel/src/main/java/org/eclipse/syson/sysml/helper/MembershipComputer.java

```diff
@@ -0,0 +1,222 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.syson.sysml.helper;
+
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.Collectors.toSet;
+
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Stream;
+
+import org.eclipse.emf.common.util.BasicEList;
+import org.eclipse.emf.common.util.EList;
+import org.eclipse.emf.ecore.InternalEObject;
+import org.eclipse.emf.ecore.util.EcoreEList;
+import org.eclipse.syson.sysml.Conjugation;
+import org.eclipse.syson.sysml.Element;
+import org.eclipse.syson.sysml.Import;
+import org.eclipse.syson.sysml.Membership;
+import org.eclipse.syson.sysml.MembershipImport;
+import org.eclipse.syson.sysml.Namespace;
+import org.eclipse.syson.sysml.NamespaceImport;
+import org.eclipse.syson.sysml.Specialization;
+import org.eclipse.syson.sysml.SysmlPackage;
+import org.eclipse.syson.sysml.Type;
+import org.eclipse.syson.sysml.Usage;
+import org.eclipse.syson.sysml.VisibilityKind;
+
+/**
+ * Object in charge of computing:
+ * <ul>
+ * <li>visibleMemberships</li>
+ * <li>inheritedMemberships</li>
+ * <li>importedMemberships</li>
+ * </ul>
+ * 
+ * @author Arthur Daussy
+ */
+public class MembershipComputer<T extends Element> {
+
+    private Set<Element> visited;
+
+    private T sourceElement;
+
+    public MembershipComputer(T sourceElement, EList<? extends Namespace> excluded) {
+        super();
+
```

Comment:
Please remove empty line (and check for other empty lines in this PR, especially at the begin/end of a method or at the end of a class)